credmark/cmf/types/pool.py

```diff
@@ -0,0 +1,65 @@
+from typing import List, Union
+from credmark.cmf.model.errors import ModelDataError
+from credmark.cmf.types.token import Tokens, Token
+from credmark.cmf.types.contract import Contract
+from credmark.dto import DTOField, IterableListGenericDTO, PrivateAttr
+
+
+class Pool(Contract):
+    """
+    A pool is a contract that consolidates assets on behalf of many accounts to perform
+    a financial function.
+    """
+
+    tokens: Tokens = DTOField(description="The tokens that are stored in the pool.")
+    balances: List[float] = DTOField(description="The balances that are stored in the pool.")
```

Comment:
An open question is do we want a Pool object to have a list of positions that get more attributes as they get more specific?

Reply:
@iwyrkore @leafyoung

Reply:
I would think so. Shouldn't it just be a list of Positions instead of separate lists of tokens and balances?

Reply:
I would prefer this flat structure of lists of value, instead of embedding object. This would make it easier to be converted to a pandas' data frame.

Reply:
The problem is keeping the lists in sync (matching indices) and adding more attributes as @newtectonics mentioned. Also, even looping through them you need to do a zip or more complicated loop handling.

Reply:
@newtectonics This seems to be a Contract containing a Portfolio which is a list of Positions (amount (float) and asset (Token)). Should we not reuse those DTOs here?